### PR TITLE
feat: Added a footer [#2]

### LINKED ISSUE #2: Add a footer
Add a footer that acts a site map, and includes some information about the pod, such as:

- Blog
- About
- Socials
- Projects

### PR DESCRIPTION
**Pull request by yyjlincoln**
# Feature: Added a footer

## Intuition
A footer plays an important part in web design. It can serve as a sitemap that provides easy navigation between pages.

## Implementation
- We added new files, including `footer.html` and `footer.scss` to construct the UI.
- We added references to `footer.html` in `index` and `default` so that every page has the footer.

## Implementation Notes
- A body tag is automatically generated when the page is rendered (although it's unclear whether it was by the browser or by Jekyll); hence body tags in `default.html` and `index.html` are replaced with `<div class="main-content">`
- To provide a better visual, we introduced a minimum height of `100vh` for elements with class `.main-content`.

## Authors
- @yyjlincoln and @Dhruv-Sachdev1313

## PR Checklist
- [x] My Pod Leader knows I'm working on this Pull Request
- [x] I've explained what the Pull Request is adding.
- [x] I've explained why this is important.

## Additional Info
This pull request closes #2.

### CODE REVIEW

File: _layouts/default.html [hunk 1]
```diff
@@ -2,10 +2,15 @@
 ---
 <!DOCTYPE html>
 <html lang="en">
+
 {% include header.html %}
 
-<body>
-    {{ content }}
-</body>
+<div class="main-content">
```
Comment (ditsuke) [inline _layouts/default.html:8]: How about `<main>`
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main

**Review by ditsuke — COMMENTED** [review]
LGTM! Left a comment about using `<main>`, more semantic but nothing serious.

File: _layouts/default.html (hunk 1 above)
Comment (yyjlincoln) [reply _layouts/default.html:8]: That sounds better. I'll do it now.

Comment (yyjlincoln) [reply _layouts/default.html]: Done! Feel free to approve the PR if you think it's ready to go.

Comment (ditsuke) [reply _layouts/default.html]: Awesome!

**Review by ditsuke — APPROVED** [review]

**Review by Aniket762 — COMMENTED** [review]
Looks great!

**Review by Dhruv-Sachdev1313 — APPROVED** [review]
Looks good to me!

File: _layouts/page.html [hunk 2]
```diff
@@ -6,7 +6,7 @@
     <div class="container">
         <div class="logo">
             <a href="/">
-                <img src="/assets/img/logo.svg"/>
+                <img src="/assets/img/logo.svg" />
```
Comment (gmelodie) [inline _layouts/page.html:9]: :scissors:

File: _layouts/default.html [hunk 3]
```diff
@@ -2,10 +2,15 @@
...
 {% include header.html %}
 
-<body>
-    {{ content }}
-</body>
+<main>
+    <div>
+        {{ content }}
+    </div>
+</main>
```
Comment (gmelodie) [inline _layouts/default.html:12]: As this is not the purpose of this PR, leave it as it was

File: _layouts/page.html [hunk 4]
```diff
@@ -17,4 +17,5 @@
 
 <div class="container page">
     {{ content }}
-</div>
\ No newline at end of file
+</div>
+
```
Comment (gmelodie) [inline _layouts/page.html:21]: :scissors:

File: _sass/all.scss [hunk 5]
```diff
@@ -55,6 +56,10 @@ a {
     color: black;
 }
 
+main {
+  min-height: 100vh;
+}
+
```
Comment (gmelodie) [inline _sass/all.scss:62]: Change `main` to `body`

File: index.html [hunk 6]
```diff
@@ -2,16 +2,24 @@
 ---
 <!DOCTYPE html>
 <html lang="en">
+
```
Comment (gmelodie) [inline index.html:5]: :scissors:

File: index.html [hunk 7]
```diff
@@ -2,16 +2,24 @@
...
+        {% include profile.html %}
+
+        {% include description.html %}
+
+        {% include projects.html %}
+
+    </div>
+
+</main>
```
Comment (gmelodie) [inline index.html:21]: Again, leave it as it was before

**Review by gmelodie — CHANGES_REQUESTED** [review]
A PR should only fix the issue at hand, no more, no less. Changing more than what's required might cause unexpected bugs and merge conflicts and plus these other changes were not discussed with the rest of the team.

File: _sass/all.scss (hunk 5 above)
Comment (yyjlincoln) [reply _sass/all.scss:62]: We tried. I believe Jekyll automatically generates the body tag that includes the footer element, so it did not achieve what we wanted. We'll give it another try.

**Comment by yyjlincoln** [comment]
Fixed up the history commit messages just now & formatting to minimize changes to main.
Merging to `main`:
- all checks have passed
- at least two approvals
- fixed up the suggested changes

**Pull request merged**